algebra/Atiyah.lean

```diff
@@ -0,0 +1,1695 @@
+      specialize hu (λ T₁ T₂ ⟨⟨t₁, ht1⟩, htc1, hts1⟩ ⟨⟨t₂, ht2⟩, htc2, hts2⟩,
+        begin
+          specialize hc t₁ ht1 htc1 t₂ ht2 htc2,
+          rw [← hts1, ← hts2] at *,
+          cases classical.em (subtype.mk t₁ ht1 = subtype.mk t₂ ht2) with ht12 ht12,
+          have := subtype.mk.inj ht12,
+          rw set.set_eq_def at this,
+          left, exact (λ x hx, (this x).1 hx),
+          specialize hc ht12,
+          exact hc
+        end),
+      let ub : ↥(ideals_not_univ α) := ⟨⋃₀ U, hu⟩,
+      existsi ub,
+      intros T htc x hx,
+      cases T with T ht,
+      exact ⟨T, ⟨⟨T, ht⟩, htc, rfl⟩, hx⟩,
+      let ub : ↥(ideals_not_univ α) := ⟨zero_ideal α, zero_ideal.is_ideal α, by simpa using hz.symm⟩,
+      existsi ub,
+      intros T htc,
+      exfalso,
+      exact h ⟨T, htc⟩
+    end),
+  specialize z (λ A B C hab hbc x hx, hbc $ hab hx),
+  cases z with m z,
+  cases m with m hm1,
+  cases hm1 with h1 h2,
+  existsi m,
+  existsi h1,
+  constructor,
+  intro h, apply h2, rw h, trivial,
+  intros T _ ht,
+  cases classical.em ((1:α) ∈ T),
+  right, exact is_ideal.univ_of_one_mem T h,
+  specialize z ⟨T, _inst_3, h⟩,
+  specialize z ht,
+  left, exact set.eq_of_subset_of_subset z ht
+end
+
+end prop_1_3
+
+-- Proposition 1.3 end
+
+
+-- Corollary 1.4 start
+
+theorem ideals_not_univ.to_maximal_ideal {α : Type u} [comm_ring α] (S : set α) [is_ideal α S] :
+S ∈ ideals_not_univ α → ∃ (m : set α) (hm : is_ideal α m), @is_maximal_ideal _ _ m hm ∧ S ⊆ m :=
+begin
+  intro h,
+  cases h with h1 h2,
+  have z := zero_ne_one.to_maximal_ideal (α/S),
+  specialize z (λ h, h2 $ is_ideal.mem_of_coset_zero S 1 h.symm),
+  cases z with m hm,
+  cases hm with h1 h2,
+  existsi quotient_to_ideal S m,
+  existsi quotient_to_ideal.is_ideal S m,
+  have hsm : S ⊆ quotient_to_ideal S m,
+    intro x,
+    simp [is_ideal.mem_iff_coset_zero S,is_ideal.to_coset],
+    intro hx,
+    rw hx,
+    apply is_ideal.zero_mem,
+  split,
+  cases h2,
+  constructor,
+  intro h,
+  apply h2_ne_univ_ideal,
+  rw [set.set_eq_def] at h,
+  apply set.ext,
+  intro x,
+  cases is_ideal.coset_rep x with y hy,
+  rw ← hy,
+  specialize h y,
+  simp at *,
+  exact h,
+  intros T _ ht,
+  have hst : S ⊆ T := set.subset.trans hsm ht,
+  specialize h2_no_between (ideal_to_quotient S T),
+  have ht2 := ideal_to_quotient.subset S _ _ ht,
+  rw quotient_to_contains_to_quotient at ht2,
+  specialize h2_no_between ht2,
+  cases h2_no_between with h h,
+  left,
+  apply set.eq_of_subset_of_subset,
+  rw set.subset.antisymm_iff at h,
+  have h3 := quotient_to_ideal.subset S _ _ h.1,
+  rw contains_to_quotient_to_contains at h3,
+  exact h3,
+  exact hst,
+  exact ht,
+  right,
+  apply set.ext,
+  rw set.set_eq_def at h,
+  intro x,
+  specialize h (is_ideal.to_coset S x),
+  simp at *,
+  cases h with y hy,
+  cases hy with hy1 hy2,
+  calc x = y - (y - x) : by norm_num
+     ... ∈ T : is_ideal.sub_mem hy1 (hst hy2),
+  exact hsm
+end
+
```

Comment:
In 1.1 you proved that the bijection between ideals containing S and ideals of alpha/S is order-preserving. Can you somehow use this to shorten this proof? Alternatively can you use that (alpha/S)/(m/S) is canonically alpha/m (an extremely useful lemma, it's basically Prop 2.1 in A-M but then you need to check the ring structures match), so one is a field iff the other is.